plotting/plot_handler.py

```diff
@@ -0,0 +1,107 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Implements the PlotHandler class which acts as the controller for the plot functionality.
+This class is (will be) responsible for:
+Searching for and retrieving an existing plotting file via the SFTP client
+Getting the associated plotting meta data file
+Instructing the Plotting factory to build an IFrame based on the above
+"""
+import logging
+from utils.clients.sftp_client import SFTPClient
+
+LOGGER = logging.getLogger('app')
+
+
+# pylint:disable=line-too-long, no-else-return
+class PlotHandler:
+    """
+    Utility that takes parameters for a run and (for now) checks if an associated image exists and retrieves it.
+    :param instrument_name: The name of the beamline/spectrometer/instrument.
+    :param rb_number: The ISIS RB number.
+    :param run_number: The run number on the given instrument for the given RB number.
+    :param plot_type:
+        The type of plot file / file extension to be searched for (e.g. "png"). Can be a list of extensions.
+        If None, a list of common graphics file extensions is searched for.
+    """
+
+    def __init__(self, instrument_name, rb_number, run_number, plot_type=None):
+        self.instrument_name = instrument_name
+        self.rb_number = rb_number
+        self.run_number = run_number
+        if plot_type is None:
+            # list of plot types the handler looks for if none is provided; it searches for the types in the order of
+            # the list and selects the first one that exists. In future: could add vector graphics (.pdf, .eps)
+            self.plot_type = ["png", "jpg", "bmp", "gif", "tiff"]
```

Comment:
Since `self.plot_type` is a list, I'd change it to `self.plot_types` (plural)
Or perhaps even `self.file_extensions` would be more accurate? Unless this can take file names as well as extensions

Reply:
Will make the suggested change. The naming was initially chosen because of issue 480 (https://github.com/ISISScientificComputing/autoreduce/issues/480), but we later realised that we don't really need a get_plot_type method.